src/components/slider/draw.js

```diff
@@ -0,0 +1,507 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var d3 = require('d3');
+
+var Plotly = require('../../plotly');
+var Plots = require('../../plots/plots');
+var Lib = require('../../lib');
+var Color = require('../color');
+var Drawing = require('../drawing');
+var svgTextUtils = require('../../lib/svg_text_utils');
+var anchorUtils = require('../legend/anchor_utils');
+
+var constants = require('./constants');
+
+
+module.exports = function draw(gd) {
+    var fullLayout = gd._fullLayout,
+        sliderData = makeSliderData(fullLayout);
+
+    // draw a container for *all* sliders:
+    var sliders = fullLayout._infolayer
+        .selectAll('g.' + constants.containerClassName)
+        .data(sliderData.length > 0 ? [0] : []);
+
+    sliders.enter().append('g')
+        .classed(constants.containerClassName, true)
+        .style('cursor', 'ew-resize');
+
+    sliders.exit().remove();
+
+    // If no more sliders, clear the margisn:
+    if(sliders.exit().size()) clearPushMargins(gd);
+
+    // Return early if no menus visible:
+    if(sliderData.length === 0) return;
+
+    var sliderGroups = sliders.selectAll('g.' + constants.groupClassName)
+        .data(sliderData, keyFunction);
+
+    sliderGroups.enter().append('g')
+        .classed(constants.groupClassName, true);
+
+    sliderGroups.exit().each(function(sliderOpts) {
+        d3.select(this).remove();
+
+        Plots.autoMargin(gd, constants.autoMarginIdRoot + sliderOpts._index);
+    });
+
+    // Find the dimensions of the sliders:
+    for(var i = 0; i < sliderData.length; i++) {
+        var sliderOpts = sliderData[i];
+        findDimensions(gd, sliderOpts);
+    }
+
+    sliderGroups.each(function(sliderOpts) {
+        computeLabelSteps(sliderOpts);
+
+        drawSlider(gd, d3.select(this), sliderOpts);
+
+        makeInputProxy(gd, d3.select(this), sliderOpts);
+
+    });
+};
+
+function makeInputProxy(gd, sliderGroup, sliderOpts) {
+    sliderOpts.inputProxy = gd._fullLayout._paperdiv.selectAll('input.' + constants.inputProxyClass)
+        .data([0]);
```

Comment:
This is not currently implemented. The idea is that it creates an `<input type="range">` that captures focus (can it actually do that programatically?) so that accessibility things like keyboard arrows to change the slider position actually work. Just an idea. Might need to go.